fcrepo-indexing-solr/src/main/java/org/fcrepo/camel/indexing/solr/SolrRouter.java

```diff
@@ -180,6 +187,12 @@ public void configure() throws Exception {
                 .removeHeaders("CamelHttp*")
                 .setHeader(HTTP_METHOD).constant("POST")
                 .setHeader(HTTP_QUERY).simple("commitWithin=" + config.getCommitWithin())
+                .choice()
+                .when(useSolrAuth)
+                    .process(new AddBasicAuthProcessor(solrUsername, solrPassword))
+                    .log(LoggingLevel.DEBUG, logger, "Authenticating solr with: " + solrUsername + ":" + solrPassword)
+                .otherwise()
+                    .log(LoggingLevel.DEBUG, logger, "No Solr Auth provided")
```

Comment:
This `choice()` block is unnecessary as the `AddBasicAuthProcessor()` already checks that the provided username is not blank before applying the header to the request. See in the [TriplestoreRouter](https://github.com/fcrepo-exts/fcrepo-camel-toolbox/blob/main/fcrepo-indexing-triplestore/src/main/java/org/fcrepo/camel/indexing/triplestore/TriplestoreRouter.java#L133-L134) and the actual [AddBasicAuthProcessor](https://github.com/fcrepo-exts/fcrepo-camel-toolbox/blob/main/fcrepo-camel-common/src/main/java/org/fcrepo/camel/common/processor/AddBasicAuthProcessor.java#L36)

Reply:
Due to the above not working, I'd say to just remove this `choice` and leave it as 
```java
.process(new AddBasicAuthProcessor(solrUsername, solrPassword))
.log(LoggingLevel.DEBUG, logger, "Authenticating solr with: " + solrUsername)
```
Also removing the password so we don't log it.